src/System.Private.CoreLib/shared/System/Globalization/TimeSpanFormat.cs

```diff
@@ -105,10 +94,166 @@ private static StringBuilder FormatToBuilder(TimeSpan value, ReadOnlySpan<char>
             return FormatCustomized(value, format, DateTimeFormatInfo.GetInstance(formatProvider), result: null);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsFormatC(ReadOnlySpan<char> format) =>
+            format.Length == 0 ||
+            (format.Length == 1 && (format[0] == 'c' || (format[0] | 0x20) == 't'));
+
+        internal static string FormatC(TimeSpan value)
+        {
+            Span<char> destination = stackalloc char[26]; // large enough for any "c" TimeSpan
+            TryFormatC(value, destination, out int charsWritten);
+            return new string(destination.Slice(0, charsWritten));
+        }
+
+        private static bool TryFormatC(TimeSpan value, Span<char> destination, out int charsWritten)
+        {
+            // First, calculate how large an output buffer is needed to hold the entire output.
+            int requiredOutputLength = 8; // start with "hh:mm:ss" and adjust as necessary
+
+            uint fraction;
+            ulong totalSecondsRemaining;
+            {
+                // Turn this into a non-negative TimeSpan if possible.
+                long ticks = value.Ticks;
+                if (ticks < 0)
+                {
+                    requiredOutputLength = 9; // requiredOutputLength + 1 for the leading '-' sign
+                    ticks = -ticks;
+                    if (ticks < 0)
```

Comment:
the second `if (ticks < 0)` is always false, isn't it?

if `ticks` is less than a zero then `ticks = -ticks` make it more than zero.

Reply:
> the second if (ticks < 0) is always false, isn't it?

No.  -long.MinValue == long.MinValue.

Reply:
Thanks for an answer. Pretty interesting that the same applies to `int.MinValue == -int.MinValue`, but not to `sbyte` and `short`.

Reply:
That is because all arithmetic operations on types smaller than `int` are automatically promoted to `int` beforehand, see [here](https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#numeric-promotions).